packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -851,6 +851,139 @@ void main() {
     expect(tester.getTopLeft(find.text('Item 0')).dx, 72);
     expect(tester.getTopLeft(find.text('Item 1')).dx, 72);
   });
+
+  test('PopupMenuButton takes only child or only icon', () {
+    // [PopupMenuButton] should assert that either child == null, icon == null,
+    // or that both are null because only one of the two can be displayed/used.
+    expect(() => PopupMenuButton<int>(
+      itemBuilder: (BuildContext context) => <PopupMenuItem<int>>[],
+      child: Container(),
+      icon: const Icon(Icons.error),
+    ), throwsAssertionError);
+  });
+
+  testWidgets('PopupMenuButton default tooltip', (WidgetTester tester) async {
+    // If the tooltip parameter for [PopupMenuButton] is not specified,
+    // a default tooltip should be used instead.
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Column(
+            // The default tooltip should be present when
```

Comment:
This comment feels redundant, since the test name and the following comments testing each particular case shares enough detail